Gems/Terrain/Code/Tests/TerrainSurfaceGradientListTests.cpp

```diff
@@ -37,81 +31,80 @@ namespace UnitTest
             appDesc.m_stackRecordLevels = 20;
 
             m_app.Create(appDesc);
-
-            CreateEntities();
         }
 
         void TearDown() override
         {
-            m_gradientEntity2.reset();
-            m_gradientEntity1.reset();
-            m_entity.reset();
-
             m_app.Destroy();
         }
 
-        void CreateEntities()
+        AZStd::unique_ptr<AZ::Entity> CreateEntity()
         {
-            m_entity = AZStd::make_unique<AZ::Entity>();
-            ASSERT_TRUE(m_entity);
+            auto entity = AZStd::make_unique<AZ::Entity>();
+            entity->Init();
+            return entity;
+        }
 
-            m_entity->Init();
+        UnitTest::MockTerrainLayerSpawnerComponent* AddRequiredComponentsToEntity(AZ::Entity* entity)
+        {
+            auto layerSpawnerComponent = entity->CreateComponent<UnitTest::MockTerrainLayerSpawnerComponent>();
+            m_app.RegisterComponentDescriptor(layerSpawnerComponent->CreateDescriptor());
 
-            m_gradientEntity1 = AZStd::make_unique<AZ::Entity>();
-            ASSERT_TRUE(m_gradientEntity1);
+            return layerSpawnerComponent;
+        }
+    };
 
-            m_gradientEntity1->Init();
+    TEST_F(TerrainSurfaceGradientListTest, SurfaceGradientMissingRequirementsActivateFails)
+    {
+        auto entity = CreateEntity();
 
-            m_gradientEntity2 = AZStd::make_unique<AZ::Entity>();
-            ASSERT_TRUE(m_gradientEntity2);
+        auto terrainSurfaceGradientListComponent = entity->CreateComponent<Terrain::TerrainSurfaceGradientListComponent>();
+        m_app.RegisterComponentDescriptor(terrainSurfaceGradientListComponent->CreateDescriptor());
 
-            m_gradientEntity2->Init();
-        }
+        const AZ::Entity::DependencySortOutcome sortOutcome = entity->EvaluateDependenciesGetDetails();
+        EXPECT_FALSE(sortOutcome.IsSuccess());
 
-        void AddSurfaceGradientListToEntities()
-        {
-            m_layerSpawnerComponent = m_entity->CreateComponent<UnitTest::MockTerrainLayerSpawnerComponent>();
-            m_app.RegisterComponentDescriptor(m_layerSpawnerComponent->CreateDescriptor());
+        entity.reset();
```

Comment:
Same in this file: `entity` is now a local variable, `entity.reset();` could be removed from all the tests.

Reply:
Fixed